### PR TITLE
`e5-mistral` evaluation via vLLM

### PR DESCRIPTION
**Pull request by robertgshaw2-neuralmagic**
- Launch server:

```bash
vllm serve intfloat/e5-mistral-7b-instruct
```

- Run eval script:

```bash
python3 run-e5.py
```

### CODE REVIEW

File: run-e5.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import asyncio
+
+import numpy as np
+from typing import Any
+from openai import AsyncOpenAI
```
Comment (Samoed) [inline run-e5.py:5]: You should move imports of external library inside the class to make it optional

File: run-e5.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+model = AsyncOpenAIWrapper(model_name=MODEL)
+model.mteb_model_meta = ModelMeta(
+    name=MODEL,
+    revision=None,
+    release_date=None,
+    languages=None,
+    similarity_fn_name=None,)
+
+tasks = mteb.get_tasks(tasks=TASKS)
```
Comment (Samoed) [inline run-e5.py:88]: When this file will be imported it will try to evaluate model. I think this should be removed

**Review by KennethEnevoldsen — COMMENTED** [review]
Very happy to see this PR! We already have an implementation of e5 mistral. However, I assume this is notably faster, but it would be nice we an actual comparison of both scores and speed.

We also implement models in `mteb/models`. I believe we could do this here as well. 

I assume it produces different scores (given vLLM) so it might be worth changing the name (related to #1211), but we can take that after a test.

**Comment by robertgshaw2-neuralmagic** [comment]
> Very happy to see this PR! We already have an implementation of e5 mistral. However, I assume this is notably faster, but it would be nice we an actual comparison of both scores and speed.
> 
> We also implement models in `mteb/models`. I believe we could do this here as well.
> 
> I assume it produces different scores (given vLLM) so it might be worth changing the name (related to #1211), but we can take that after a test.

Hey @KennethEnevoldsen - apologies, I did not meant to submit this PR onto upstream it was just a simple POC for myself to eval the embedding models running in vLLM. I managed to match the scores running through GritLM on a few STS tasks. I made this to automate correctness testing in vLLM

If this is something you would like to see in MTEB, I will clean it up

**Comment by KennethEnevoldsen** [comment]
> If this is something you would like to see in MTEB, I will clean it up

If you can get it to match the performance at a faster speed I would love to have it added (cc @Muennighoff; it might be worth integrating this in GritLM)

**Comment by KennethEnevoldsen** [comment]
Will close this issue as it seems to have gotten stale, though I believe this is a great PR so do feel free to re-open it or add a PR to GritLM.

**Pull request closed without merging**